pretyping/inferCumulativity.ml

```diff
@@ -155,6 +156,7 @@ and infer_stack infos variances (stk:CClosure.stack) =
         infer_vect infos variances (Array.map (mk_clos e) br)
       | Zshift _ -> variances
       | Zupdate _ -> variances
+      | Zprimitive _ -> variances (* FIXME check this *)
```

Comment:
@SkySkimmer how do I know what to put here?

Reply:
This asks "when is zprimitive foo convertible to itself with just universes changed".
You basically have to replace `f`/`ccnv` in Reduction.convert_stacks by `infer_term` (order of folds shouldn't matter (except for perf but I don't know of any code stressing infercumulativity so we can't test which order is better)).
Going by the code in reduction.ml around https://github.com/maximedenes/coq/blob/c1ed789fd136853a6a0e59de26e5a4cbbe535494/kernel/reduction.ml#L683-L687 that would mean
```suggestion
      | Zprimitive (_,_,rargs,kargs) ->
        let variances = List.fold_left (fun variances c -> infer_fterm CONV infos variances c []) variances rargs in 
        let variances = List.fold_left (fun variances (_,c) -> infer_fterm CONV infos variances c []) variances kargs in 
        variances
```